src/macaron/slsa_analyzer/ci_service/jenkins.py

```diff
@@ -56,7 +81,111 @@ def build_call_graph(self, repo_path: str, macaron_path: str = "") -> CallGraph:
         CallGraph : CallGraph
             The call graph built for the CI.
         """
-        return CallGraph(BaseNode(), "")
+        if not macaron_path:
+            macaron_path = global_config.macaron_path
+
+        root: BaseNode = BaseNode()
+        call_graph = CallGraph(root, repo_path)
+
+        # To match lines that start with sh '' or sh ''' ''' (either single or triple quotes)
+        # we need to account for both the single-line and multi-line cases.
+        pattern = r"^\s*sh\s+'{1,3}(.*?)'{1,3}$"
+        workflow_files = self.get_workflows(repo_path)
+
+        for workflow_path in workflow_files:
+            try:
+                with open(workflow_path, encoding="utf-8") as wf:
+                    lines = wf.readlines()
+            except OSError as error:
+                logger.debug("Unable to read Jenkinsfile %s: %s", workflow_path, error)
+                return call_graph
+
+            # Add internal workflow.
+            workflow_name = os.path.basename(workflow_path)
+            workflow_node = JenkinsNode(
+                name=workflow_name,
+                node_type=JenkinsNodeType.INTERNAL,
+                source_path=workflow_path,
+                caller=root,
+            )
+            root.add_callee(workflow_node)
+
+            # Find matching lines.
+            for line in lines:
+                match = re.match(pattern, line)
```

Comment:
If we are matching the Jenkinsfile one line at a time, would we miss multi-lines cases? For example
```
sh '''
    echo "Multiline shell steps works too"
    ls -lah
'''
```
Example taken from https://www.jenkins.io/doc/pipeline/tour/running-multiple-steps/#linux-bsd-and-mac-os.

Or are we not supporting multi-line cases at the moment.

Reply:
No, the regex only supports a single-line shell scripts. I will add a TODO to support multi-line scripts.